Add 'git: view commit' command palette action

schpet opened the pull request:
adds a 'git: view commit' – accepting a ref (e.g. HEAD, an sha, etc) to more easily navigate to the git commit view.

<img width="3024" height="1888" alt="Screenshot 2025-09-26 at 21 43 09@2x" src="https://github.com/user-attachments/assets/c001baec-66c2-46e5-b4a7-f691631f4166" />


if a bad ref is entered, the user is shown a generic error

<img width="2734" height="1442" alt="Screenshot 2025-09-27 at 21 04 52@2x" src="https://github.com/user-attachments/assets/abdbd92d-ef0b-4de9-afb9-e9e52607dfdd" />

happy to adjust any of that. also worth noting is the `git: branch` command UI is a bit nicer, can e.g. show you some metadata on the commit before you select it, so happy to take it further in that direction if desired, but thought i'd keep it simple to start. 

Release Notes:

- Added view commit command palette action

schpet committed:
Add view commit command pallet action

cole-miller committed:
Merge remote-tracking branch 'origin/main' into push-rwqultwkrknv

cole-miller commented:
Thanks! This seems useful, happy to merge as-is. If you're interested in taking it further, I like the idea of making it possible to see the commit metadata before opening it (and that would also give an affordance for showing the user that their query isn't valid immediately, instead of just showing a toast after they hit confirm).

Angelk90 commented:
@schpet : In the future it would be nice to be able to suggest commits.

For example:
- I enter a few characters of the reference commit that are in sequence, and it suggests the commit if it exists.
For example, I enter the first n characters and it tells me which commits satisfy the query.
- I enter the title of a commit and it suggests the commits that satisfy the query.

schpet committed:
add commit metadata preview to ref picker modal

display author, subject, and relative timestamp as the user types a
git ref, with debouncing to avoid excessive lookups.

schpet commented:
as a follow up to this, i'm interested in URI support for the git commit view, i.e. supporting something along the lines of `zed://git/path/to/repo#sha`

this would allow customize the output of my git (or jj) log to use [OSC 8 links](https://github.com/Alhadis/OSC8-Adoption) to link git shas straight into zed – letting me use version control in the terminal, and review a diff of code changes in zed.

basically i'm finding myself in this situation

- 📈 me looking at diffs
- 📉 me looking at the working copy

which is reducing my time in zed, a place i like to be!

ChristopherBiscardi committed:
Merge remote-tracking branch 'upstream' into push-rwqultwkrknv

ChristopherBiscardi committed:
re-pub test_new

ChristopherBiscardi committed:
fmt

ChristopherBiscardi committed:
Merge remote-tracking branch 'upstream' into push-rwqultwkrknv

ChristopherBiscardi commented:
I've merged main and updated the PR to be current, as well as tested the functionality. The test currently fails for an unrelated-to-the-functionality reason: 

> called `Result::unwrap()` on an `Err` value: no state of type theme::theme_settings_provider::GlobalThemeSettingsProvider exists

After the test is fixed, we can get this merged.

ChristopherBiscardi committed:
fix test using newer theme_settings functions

Co-authored-by: Marshall Bowers <git@maxdeviant.com>

maxdeviant reviewed (approved):
Thank you!

schpet commented:
whoa, very cool. thanks all!
